articles/libraries/lock/v11/migration-angularjs-v9.md

```diff
@@ -4,7 +4,192 @@ title: Migrating AngularJS Applications to from Lock v9 to Lock v11
+# Migrating AngularJS Applications from Lock v9 to v11
 
-TODO.
+This guide will help you migrating your Angular 1.x application from Lock 9 to Lock 11. 
 
+<%= include('../../_includes/_get_lock_latest_version') %>
+
+## 2. Adding Angular Lock Library
+
+Most Angular 1.x apps used the [auth0-angular](https://www.npmjs.com/package/auth0-angular) which is currently deprecated. This guide will use [auth0-lock](https://www.npmjs.com/package/auth0-lock) 3.0 to simplify using Lock 11.
+
+Use npm to uninstall the old library and install the new one.
+
+```
+npm uninstall auth0-angular --save
+npm install angular-lock --save
+```
+
+Then include it in your `index.html` file:
+
+```html
+<!-- index.html -->
+<script src="node_modules/angular-lock/dist/angular-lock.min.js"></script>
+```
+
+## 3. Updating App File
+
+In your `app.js` file:
+
+* Add `auth0.lock` to your Angular module
+* Inject `lockProvider` in your configuration
+* Initialize a Lock instance in `config()`
+* Inject your authentication service in `run()` and call the method that listens for authentication events (we will implement changes in the authentication service next)
+
+```js
+// app.js
+(function() {
+  'use strict';
+
+  angular
+    .module('app', ['ngRoute', 
+//   'auth0' 
+     'auth0.lock']) // Add auth0.lock
+      .config(config)
+      .run(run);
+
+  config.$inject = [
+    '$locationProvider',
+    '$routeProvider',
+ // 'authProvider',
+    'lockProvider' // Inject lockProvider
+  ];
+
+  function config(
+    $locationProvider,
+    $routeProvider,
+//  authProvider
+    lockProvider // Provide lockProvider
+  ) {
+    $routeProvider
+      .when('/', {
+        controller: 'HomeController',
+        templateUrl: 'app/home/home.html',
+        controllerAs: 'vm'
+      })
+      .otherwise({
+        redirectTo: '/'
+  });
+
+  $locationProvider.hashPrefix('');
+  $locationProvider.html5Mode(true);
+
+  // Configure Auth0 Lock instance
+  // Read more about configuration here:
+  // https://auth0.com/docs/libraries/lock/v11
+  lockProvider.init({
+     domain: '${account.namespace}'),
+     clientID: '${account.clientId}',
+     options: {
+       autoclose: true,
+       auth: {
+         responseType: 'token id_token',
+         audience: 'https://' + '${account.namespace}' + '/userinfo',
+         params: {
+           scope: 'openid profile email'
+         }
+       }
+     }
+   });
+  }
+
+  // Inject authentication service and
+  // run method to handle authentication
+  run.$inject = ['authService'];
+
+  function run(authService) {
+    authService.handleAuthentication();
+  }
+
+}());
+```
+
+## 4. Updating Authentication Service
+
+If you are using an authentication service, you will need to make a few minor changes there as well. Lock 11 provides events so you can execute functionality when the user is authenticated, when there is an authentication error, etc. You can read more at the [Lock 11 documentation](sdk).
```

Comment:
@aaguiarz Is `sdk` meant to be `/libraries/lock/v10`?

Reply:
Yes, just fixed it